File: routes/outlier.js
var express = require('express');
var router = express.Router();
var summary = require('../lib/summary.js')
var database = require('../lib/DBConnecter.js');

router.get('/', function (req, res, next) {
  var minArray = req.query.minArray;
  var maxArray = req.query.maxArray;

  var uid = req.query.uid;
  const db = new database(uid);
  db.load('tmp', function (data) {
    df = new DataFrame(data)
    //selected variable
    var variable = new Array();
    seqNames = df.columns;
    //variable
    var variable = new Array();
    seqNames.forEach(function (element) {
      variable.push(element);
    });
    if (minArray != null) {
      minArray.forEach(function (element, index) {
        if (element > 0) {
          df = df.filter(df.get(variable[index]).gt(element));
        }
      });
    }
    if (maxArray != null) {
      maxArray.forEach(function (element, index) {
        if (element > 0) {
          df = df.filter(df.get(variable[index]).lt(element));
        }
      });
    }
    db.save(df.to_json({ orient: 'records' }))

    var data = summary(df)
    res.json({ data: data })
  });

});
router.get('/modal', function (req, res, next) {

  var uid = req.query.uid;
  // console.log("uid = "+uid)
  const db = new database(uid);
  db.load('tmp', function (data) {
    df = new DataFrame(data)
    seqNames = df.columns;

    var variable = new Array();
    var data = new Array();
    seqNames.forEach(function (element) {
      variable.push(element);
      data.push(df.get(element).to_json({ orient: 'records' }))
    });

    res.json({ variable: variable, data: data })
  });


});

module.exports = router;

File: apiserver/server/router/processor/outlier.py
import pandas as pd
import numpy
from ... import firebase_init
from django.http import HttpResponse, JsonResponse
from django.views.decorators.csrf import csrf_exempt
import json


@csrf_exempt
def process(request):
    db = firebase_init.db
    body = json.loads(request.body.decode('utf-8'))
    uid = body['uid']
    minArray = body['minArray']
    maxArray = body['maxArray']
    seqNames = body['variableArray']
    ref = db.reference("/" + uid + "/tmp")
    snapshot = ref.get()
    data = snapshot['data']

    dataList = []
    for key, value in data.items():
        dataList.append(value)


    postData = pd.DataFrame(dataList)

    colName = postData.columns.values.tolist()
    for idx, element in enumerate(colName):
        element = element.replace("&35;", "#")
        element = element.replace("&36;", "$")
        element = element.replace("&46;", ".")
        element = element.replace("&91;", "[")
        element = element.replace("&93;", "]")
        element = element.replace("&47;", "/")
        colName[idx] = element

    postData.columns = colName


    if minArray != None:
        for idx, element in enumerate(minArray):
            if element != None:
                postData = postData[postData.get(
                    seqNames[idx]) > float(element)]


    if maxArray != None:
        for idx, element in enumerate(maxArray):
            if element != None:
                postData = postData[postData.get(
                    seqNames[idx]) < float(element)]

    return JsonResponse({"snapshot": postData.to_json(orient='records')})
